spring-web/src/test/java/org/springframework/web/service/invoker/RequestParamArgumentResolverTests.java

```diff
@@ -57,12 +59,155 @@ void requestParam() {
+		assertThat(body).isInstanceOf(MultiValueMap.class);
+		assertThat((MultiValueMap<String, String>) body).hasSize(2)
+				.containsEntry("param1", List.of("value 1"))
+				.containsEntry("param2", collectionParams);
+	}
+
+	@Test
+	@SuppressWarnings("unchecked")
+	void requestParam6() {
+		ConversionService conversionService = new DefaultConversionService();
+		boolean formatAsSingleValue = false;
+		Service2 service =
+				HttpServiceProxyFactory.builderFor(this.client)
+						.customArgumentResolver(new RequestParamArgumentResolver(conversionService, formatAsSingleValue))
+						.build().createClient(Service2.class);
+		List<String> collectionParams = List.of("1", "2", "3");
+		service.getForm("value 1", collectionParams);
+
+		Object body = this.client.getRequestValues().getBodyValue();
+		assertThat(body).isInstanceOf(MultiValueMap.class);
+		assertThat((MultiValueMap<String, String>) body).hasSize(2)
+				.containsEntry("param1", List.of("value 1"))
+				.containsEntry("param2", collectionParams);
+	}
+
+	@Test
+	@SuppressWarnings("unchecked")
+	void requestParam7() {
+		ConversionService conversionService = new DefaultConversionService();
+		boolean formatAsSingleValue = false;
+		Service3 service =
+				HttpServiceProxyFactory.builderFor(this.client)
+						.customArgumentResolver(new RequestParamArgumentResolver(conversionService, formatAsSingleValue))
+						.build().createClient(Service3.class);
+		List<String> collectionParams = List.of("1", "2", "3");
+		service.getForm("value 1", collectionParams);
+
+		Object uriVariables = this.client.getRequestValues().getUriVariables();
+		assertThat(uriVariables).isNotInstanceOf(MultiValueMap.class)
+				.isInstanceOf(HashMap.class);
+		assertThat((HashMap<String, String>) uriVariables).hasSize(4)
+				.containsEntry("queryParam0", "param1")
+				.containsEntry("queryParam0[0]", "value 1")
+				.containsEntry("queryParam1", "param2")
+				.containsEntry("queryParam1[0]", String.join(",", collectionParams));
+	}
+
 
 	private interface Service {
 
 		@PostExchange(contentType = "application/x-www-form-urlencoded")
 		void postForm(@RequestParam String param1, @RequestParam String param2);
 
+		@GetExchange
+		void getForm(@RequestParam String param1, @RequestParam List<String> param2);
+
+		@PutExchange(contentType = "application/x-www-form-urlencoded")
+		void putForm(@RequestParam String param1, @RequestParam List<String> param2);
+
+		@PutExchange(contentType = "application/x-www-form-urlencoded")
+		void deleteForm(@RequestParam String param1, @RequestParam List<String> param2);
+
+		@PatchExchange(contentType = "application/x-www-form-urlencoded")
+		void patchForm(@RequestParam String param1, @RequestParam List<String> param2);
 	}
 
+	@HttpExchange(contentType = "application/x-www-form-urlencoded")
+	private interface Service2 {
+
+		@GetExchange
+		void getForm(@RequestParam String param1, @RequestParam List<String> param2);
+	}
```

Comment:
I don't think HTTP GET with form data is not a valid scenario. HTTP GET is not expected to have a body.

Reply:
Same as https://github.com/spring-projects/spring-framework/pull/33220#discussion_r1685286621